core/src/test/scala/integration/kafka/network/DynamicConnectionQuotaTest.scala

```diff
@@ -179,17 +179,23 @@ class DynamicConnectionQuotaTest extends BaseRequestTest {
     reconfigureServers(props, perBrokerConfig = true, (KafkaConfig.ListenersProp, newListeners))
     waitForListener("EXTERNAL")
 
+    // we need to set the initialConnectionCount earlier and pass to verifyConnectionRate method
+    // so that the race condition won't occur for the following multi-thread test cases
```

Comment:
This comment is irrelevant now. I would just say that this is the expected connection count after each run.